experimental/auth-logic/wrappers/rekor_wrapper.go

```diff
@@ -122,18 +152,107 @@ func verifyRekordLogSignature(rekordEntry *rekord.V001Entry) (*ecdsa.PublicKey,
 	if !ok {
 		return nil, fmt.Errorf("public key is not ecdsa: %v", pubKeyDecoded)
 	}
+	return ecdsaKey, nil
+}
 
-	data, err := hex.DecodeString(*rekordEntry.RekordObj.Data.Hash.Value)
+// checkInclusionProof forces a check of the inclusion proof. The method
+// (*models.LogEntryAnon).Validate(...) will check the inclusion proof in
+// LogEntryAnon.Verification, but only if it is non-empty. If it is empty
+// it will not error, so this function just throws an error if the verification
+// is empty
+func checkInclusionProof(logEntryAnon *models.LogEntryAnon, registry strfmt.Registry) error {
+	if logEntryAnon.Verification == nil {
+		return fmt.Errorf("logEntryAnon did not have inclusion proof: %v", logEntryAnon)
+	}
+	return logEntryAnon.Validate(strfmt.Default)
+}
+
+// checkEntryPubKeyMatchesExpectedKey compares the public key of the product
+// team in the Rekor log entry to the key of the product team passed as an
+// input to this wrapper. It returns an error if they are not equal
+// (or if valid keys could not be constructed)
+func checkEntryPubKeyMatchesExpectedKey(rekordEntry *rekord.V001Entry, prodTeamKeyBytes []byte) error {
```

Comment:
Yes, but it does not answer my question! 

I am not sure the public key is included in the entry just to make it self-contained. If that is the case, we could just  ignore that and directly use the product team's key to verify the signature. Would not that be clearer? 

I actually think the public key is included in the log entry to allow verifying the identity of the signer, in a general manner. So, I think it is better to explicitly reflect this in the code.

Reply:
If we check that the key in the rekor entry is the same as an expected product team key, it doesn't matter which one we use to check the signature (since they are the same key!)

We also should check that the public key in the rekor entry is the same as the product team key because if it isn't then the signature cant be verified with the key in the entry (which is suspicious because that's not how it should work), and it also suggests the party making the entry is not the one with the trusted product team key.

Reply:
I am not entirely convinced. I think we have to figure out why exactly Rekor includes the public key in the LogEntry, and what use cases they have in mind. Because, exactly as you described above, just validating the signature against the public key in the LogEntry is not enough. 

But we can come back to this in a separate PR.